processor/elasticprocessor/internal/hostmetrics/process.go

```diff
@@ -0,0 +1,105 @@
+package hostmetrics
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func addProcessMetrics(metrics pmetric.MetricSlice, dataset string) error {
+	var timestamp pcommon.Timestamp
+	var threads, memUsage, memVirtual, fdOpen int64
+	var memUtil, total, cpuTimeValue float64
+
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+		if metric.Name() == "process.threads" {
+			dp := metric.Sum().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			threads = dp.IntValue()
+		} else if metric.Name() == "process.memory.utilization" {
+			dp := metric.Gauge().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			memUtil = dp.DoubleValue()
+		} else if metric.Name() == "process.memory.usage" {
+			dp := metric.Sum().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			memUsage = dp.IntValue()
+		} else if metric.Name() == "process.memory.virtual" {
+			dp := metric.Sum().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			memVirtual = dp.IntValue()
+		} else if metric.Name() == "process.open_file_descriptors" {
+			dp := metric.Sum().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			fdOpen = dp.IntValue()
+		} else if metric.Name() == "process.cpu.time" {
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+				timestamp = dp.Timestamp()
+				value := dp.DoubleValue()
+				if state, ok := dp.Attributes().Get("state"); ok {
+					switch state.Str() {
+					case "system":
+						total += value
+					case "user":
+						total += value
+					case "wait":
+						total += value
+
+					}
+				}
+			}
+		}
+	}
+
+	memUtilPct := memUtil / 100
+	cpuTimeValue = total * 1000
```

Comment:
tiny nit: why not define both of these values here with `:=` vs one at the top and one here